src/Contacts.ts

```diff
@@ -252,10 +255,84 @@ export class Contacts {
     this.jobRunner = new JobRunner('user-preferences', client.keystore)
   }
 
+  private async validateConsentSignature(
+    signature: `0x${string}`,
+    timestampMs: number,
+    peerAddress: string
+  ): Promise<boolean> {
+    if (!signature || !timestampMs) {
+      return false
+    }
+    if (timestampMs > Date.now()) {
+      return false
+    }
+    if (timestampMs < Date.now() - 1000 * 60 * 60 * 24 * 30) {
+      return false
+    }
+    const signatureData = splitSignature(signature)
+    const message = WalletSigner.consentProofRequestText(
+      peerAddress,
+      timestampMs
+    )
+    const digest = hexToBytes(hashMessage(message))
+    // Recover public key
+    const publicKey = ecdsaSignerKey(digest, signatureData)
+    return publicKey?.getEthereumAddress() === this.client.address
+  }
+
+  private async handleConsentProofs(
+    consentProofs: {
+      consentProof: invitation.ConsentProofPayload
+      peerAddress: string
+    }[]
+  ) {
+    const validConsentProofAddresses: string[] = []
+    const validationResults = await Promise.allSettled(
+      consentProofs.map((proofItem) => {
+        return this.validateConsentSignature(
+          proofItem.consentProof.signature as `0x${string}`,
+          Number(proofItem.consentProof.timestamp),
+          proofItem.peerAddress
+        )
+      })
+    )
+    validationResults.forEach((result, index) => {
+      if (result.status === 'fulfilled' && result.value) {
+        validConsentProofAddresses.push(consentProofs[index].peerAddress)
+      }
+    })
```

Comment:
if `validateConsentSignature` becomes sync, i think you can simplify this logic with:

```
    const validConsentProofAddresses = consentProofs.reduce(
      (result, proofItem) => {
        const isValid = this.validateConsentSignature(
          proofItem.consentProof.signature as `0x${string}`,
          Number(proofItem.consentProof.timestamp),
          proofItem.peerAddress
        )
        return isValid ? result.concat(proofItem.peerAddress) : result
      },
      [] as string[]
    )
```

Reply:
Oh yes good catch, I I'll make it sync